python/mxnet/optimizer.py

```diff
@@ -299,8 +300,13 @@ def update(self, index, weight, grad, state):
         if state:
             mom = state
             mom[:] *= self.momentum
-            mom[:] += -lr * (grad + wd * weight)
-            weight[:] += mom
+            grad += wd * weight
```

Comment:
@piiswrong @tqchen There is only one line difference in nesterov vs sgd, subclassing duplicates the update code.